### DISCUSSION TITLE
Capture photo from USB camera in AVALONIA

### DISCUSSION THREAD

**Discussion by Ilesbayev-Meyrbek** [AvaloniaUI/Avalonia#9796] [General]
Has anyone taken a photo from a usb camera to Avalonium? I didn’t work in Avalon before, everything works fine on winform, but I don’t have Avalon, who can help?
I tried to use FlashCap, it detects the camera by port, but I don’t get the result at the output.
Coding in Linux OS.

----------------------------------------------------------------
    public async Task GetPic()
    {
        var devices = new CaptureDevices();

        // Open a device with a video characteristics:
        var descriptors = devices.EnumerateDescriptors().ElementAt(1);
        System.Drawing.Image bitmap;

        using var device = await descriptors.OpenAsync(
            descriptors.Characteristics[2],
            async bufferScope =>
        {
            byte[] image = bufferScope.Buffer.CopyImage();

            var ms = new MemoryStream(image);

            bitmap = Bitmap.FromStream(ms);
        });

        await device.StartAsync();
        await device.StopAsync();
    }

----------------------------------------------------------------

bitmap returns NULL.

**Comment by maxkatz6** [comment]
Avalonia is a GUI framework, it doesn't have APIs for the camera access.
You need to write platform specific code for each target platform you have. Or find a third-party library that does it already.

**Reply by Ilesbayev-Meyrbek** [reply]
I used FlashCap but the result is NULL
